### ISSUE TITLE
Fiscal year support

### ISSUE THREAD

**Issue by binodsarkar** [cube-js/cube#298]
**Fiscal year support.**
Fiscal year support in cube js. Currently "this year" means start from 1st January but sometime need fiscal year like 1st April 2019 - 31st March 2020.

**Comment by paveltiunov** [comment]
@binodsarkar I believe we should support something like `from April 1 2019 to March 31 2020`. We can think about on how to make it rolling. Are those fiscal years different for every company/country?

**Comment by binodsarkar** [comment]
> @binodsarkar I believe we should support something like `from April 1 2019 to March 31 2020`. We can think about on how to make it rolling. Are those fiscal years different for every company/country?

Yes, in most cases it is different country wise.

**Comment by walkerjono** [comment]
Normally you would have a Financial/Fiscal Calendar and the standard Gregorian Calendar (Jan->Dec) 

Examples of Financial/Fiscal Calendars:

- Jan - > Dec (US)
- July -> June (Australia)
- Apr -> Mar (NZ)

**Comment by paveltiunov** [comment]
@walkerjono Hey Jono! Thanks for posting feedback on this! What type of date ranges do you want to query? Is it `this financial year` and `last financial year`?

**Comment by walkerjono** [comment]
No problems at all. This year and last year (both Gregorian and Financial) is very common.

In addition, we normally develop financial reports with a calendar hierarchy:
Year > Half > Quarter > Period/Month > Week

Here is a very detailed example for reference:
[](https://www.sqlbi.com/articles/reference-date-table-in-dax-and-power-bi/)

**Comment by norbertbede** [comment]
we have some customers well, they have specific fiscal year. our erp has calendar definition, where we can define fiscal years, then system handle it as actual, previous. (inherit proper date from definition.)

**Comment by paveltiunov** [comment]
@walkerjono @norbertbede So what would be the best solution there? Have date ranges for all known fiscal years or parse fiscal year starting data?

Regarding dimensions there's separate issue here: https://github.com/cube-js/cube.js/issues/177.

**Comment by norbertbede** [comment]
This question looks me quite cumbersome. Probably this is at all about custom definable periods - Fiscal year is just one option, you should have specific calendars. 

Probably one option is to make own "period" cube - which is combination of: Calendar/Period/week/day. So we can join by date but you get back parent. (various levels).

Another question is relative dates - where you want make reporting/comparing 2nd month last/previous year. (another topic)

So at the end i'm not sure any change required here...
norbert

**Comment by HarshalNathe** [comment]
Can anybody get a solution to this! I'm also having different scenarios regarding fiscal year like 
``` 
- This fiscal year
- Last fiscal year
- This quarter of the fiscal year
- Last quarter of the fiscal year
- Year to date
- Last year to date
```
etc.

**Comment by InbarShirizly** [comment]
Hi, any news regarding this request (or any workaround)?
we would like to be able to change the fiscal date dynamically and query accordingly
 (we have a multi-tenant project, so each of our customers can have different fiscal date and we need to support that)

thanks!

**Comment by benhickson** [comment]
I think this could be done client-side by whatever you're querying cube with.

**Comment by igorlukanin** [comment]
Finally, this is resolved with #8488
